expr/func_aggregate.go

```diff
@@ -48,11 +48,28 @@ func (s *FuncAggregate) Exec(cache map[Req][]models.Series) ([]models.Series, er
 	out := pointSlicePool.Get().([]schema.Point)
 	s.agg.function(series, &out)
 
+	// The tags for the aggregated series is only the tags that are
+	// common to all input series
+	var commonTags map[string]string
+	if len(series[0].Tags) == 0 {
+		commonTags = make(map[string]string, 0)
+	} else {
+		commonTags = series[0].Tags
```

Comment:
any change made to `commonTags` will be reflected in `series[0].Tags`

Reply:
Is that ok? Other functions modify the input series directly as well.

Reply:
i looked through the code, turns out this was OK. because the map is set via `Series.SetTags()` and thus any changes to it are only visible in the scope of that http handler for the render request.

Reply:
I changed it to copy the tags in https://github.com/grafana/metrictank/pull/774/commits/f3da5d1fa5a9873b412eb74946128c7a20651905 anyway

Reply:
While likely true, the benchmarks show almost no difference between the approaches (since the other work greatly outweighs the cost of the map building). I like this approach, because it removes cognitive load (not having to worry about the lifetime of series tags or if other functions might need to reuse it, etc) as well as being a few lines of code smaller.

If you have strong feelings about it, I can change it back.

Reply:
ok np